pkg/action/scan.go

```diff
@@ -143,6 +130,63 @@ func recursiveScan(ctx context.Context, c Config) (*bincapz.Report, error) {
 	return r, nil
 }
 
+func processArchive(
+	ctx context.Context,
+	c Config,
+	yrs *yara.Rules,
+	r *bincapz.Report,
+	p string,
+	logger *clog.Logger,
+) error {
+	var err error
+	p, err = archive(ctx, p)
+	if err != nil {
+		return fmt.Errorf("failed to prepare archive for scanning: %w", err)
+	}
+	var ap []string
+	ap, err = findFilesRecursively(ctx, p)
+	if err != nil {
+		return fmt.Errorf("find files: %w", err)
+	}
+	for _, a := range ap {
+		err = processFile(ctx, c, yrs, r, a, logger)
+		if err != nil {
+			return err
+		}
+	}
+	if err := os.RemoveAll(p); err != nil {
```

Comment:
I moved the archive removal logic here so that the original archives aren't deleted once the scan completes.